### PR TITLE
Ability to Add a Grader to a Section

### PR DESCRIPTION
**Pull request by alexinman**
#2449 

Adds the ability to filter students by section in the Quiz>Graders tab. 
Also adds the ability to sort the Quiz Marks Spreadsheets by section.

screen shot 2016-03-19 at 1 07 48 pm
screen shot 2016-03-19 at 1 07 12 pm
screen shot 2016-03-19 at 1 07 01 pm

@david-yz-liu Is there anything else you would like done for #2449?

### CODE REVIEW

**Comment by reidka** [comment]
What I want to be able to do is to select a section to filter by, and the click on the select all checkbox, the grader I want to grade that section, and add the grader only to students in that section.

E.g. in the picture above, for Manage Graders, 
- select section c5 to display students in the section c5.
- check the checkbox next to UserName (so all the visible checkboxes turn blue), 
- check a grader, and add the grader to the students in the section.

Right now, this sequence of operations will add that grader to everyone.

**Comment by reidka** [comment]
The sorting by section in the spreadsheets doesn't work for me, even though in firebug I can see that the GradesTable has sortable set to true for the sections column.

**Comment by alexinman** [comment]
@reidka I can't replicate your issue with sorting by section in the spreadsheets

**Comment by alexinman** [comment]
The issue with clicking the select all box is due to this line (and lines like it):
https://github.com/MarkUsProject/Markus/pull/2503/files#diff-2e11ca15fd9b8b5c85d7958ab4a499d1R18

It just selects or deselects all groups/students for the table and doesn't check the current filter or the visible_rows.

I don't know how to access the current filter or the visible_rows because they are handled by `app/views/assets/javascripts/ReactComponents/table.js` and the select all checkbox is in the `app/views/graders/_graders_manager.js.jsx.erb` and the `app/views/marks_graders/_marks_graders_student_table.js.jsx.erb` files.

@reidka or @david-yz-liu do you know how I can access the visible_rows from the `table.js` class in the graders manager and marks graders student tables?

File: app/views/graders/_graders_manager.js.jsx.erb [hunk 1]
```diff
@@ -553,7 +558,8 @@
 
       // need to filter for each section
       var filters = [];
-      filters.push({name: '-', text: '-', func: function(g) { return g; } });
+      var all = '<%= j raw I18n.t('all') %>';
```
Comment (david-yz-liu) [inline app/views/graders/_graders_manager.js.jsx.erb]: Let's not use `j` and `raw` here

File: app/views/graders/_graders_manager.js.jsx.erb [hunk 2]
```diff
@@ -589,9 +595,9 @@
       }.bind(this));
 
       return (
-       <div className='tab-pane ui-tabs ui-widget ui-widget-content ui-corner-all' id='groups-tabs'>
-        <Table data={groups_data} columns={columns}
-               filters={filters} filter_type={false}
+        <div className='tab-pane ui-tabs ui-widget ui-widget-content ui-corner-all' id='groups-tabs'>
+          <Table data={groups_data} columns={columns} ref="myTable"
+               filters={filters} filter_type={filters.length>3}
```
Comment (david-yz-liu) [inline app/views/graders/_graders_manager.js.jsx.erb]: I'm not sure what the significance of the "3" is.

File: app/views/graders/_graders_manager.js.jsx.erb (hunk 1 above)
Comment (alexinman) [inline app/views/graders/_graders_manager.js.jsx.erb]: Alright.. What does j and raw do? I just copied that from somewhere else..

File: app/views/graders/_graders_manager.js.jsx.erb (hunk 2 above)
Comment (alexinman) [inline app/views/graders/_graders_manager.js.jsx.erb]: In the table.js file is suggest to use the select filter type when there's more than 3 choices to filter by.

Comment (alexinman) [inline app/views/graders/_graders_manager.js.jsx.erb]: SetFilter_type to false for simple filter type and true for the select (or fancy, I think its called) filter type

File: app/views/graders/_graders_manager.js.jsx.erb (hunk 1 above)
Comment (david-yz-liu) [inline app/views/graders/_graders_manager.js.jsx.erb]: http://apidock.com/rails/ActionView/Helpers/JavaScriptHelper/escape_javascript and http://apidock.com/rails/ActionView/Helpers/OutputSafetyHelper/raw.

Neither apply here.

File: app/views/graders/_graders_manager.js.jsx.erb (hunk 2 above)
Comment (david-yz-liu) [inline app/views/graders/_graders_manager.js.jsx.erb]: Hmm alright. Please put spaces around the `>`

**Pull request merged**